clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -330,6 +334,58 @@ static mlir::Value MakeAtomicCmpXchgValue(CIRGenFunction &cgf,
   return returnBool ? op.getResult(1) : op.getResult(0);
 }
 
+static mlir::Value MakeAtomicFenceValue(CIRGenFunction &cgf,
+                                        const CallExpr *expr,
+                                        cir::MemScopeKind syncScope) {
+  QualType typ = expr->getType();
+  auto &builder = cgf.getBuilder();
+
+  auto intType =
+      expr->getArg(0)->getType()->getPointeeType()->isUnsignedIntegerType()
+          ? builder.getUIntNTy(cgf.getContext().getTypeSize(typ))
+          : builder.getSIntNTy(cgf.getContext().getTypeSize(typ));
+
+  auto orderingVal =
+      emitToInt(cgf, cgf.emitScalarExpr(expr->getArg(1)), typ, intType);
+  auto orderingAttr =
+      orderingVal.getDefiningOp()->getAttrOfType<mlir::IntegerAttr>("value");
+
+  cir::MemOrder ordering;
```

Comment:
Probably no need for the switch: the order of the enum is already the value you are checking for, so just cast it directly:
```
// Memory order according to C++11 memory model
enum class MemOrder : uint32_t {
  Relaxed = 0,
  Consume = 1,
  Acquire = 2,
  Release = 3,
  AcquireRelease = 4,
  SequentiallyConsistent = 5,
};
```